SignalMessaging/ViewControllers/OWSNavigationController.m

```diff
@@ -0,0 +1,146 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+#import "OWSNavigationController.h"
+#import <SignalMessaging/SignalMessaging-Swift.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface UINavigationController (OWSNavigationController) <UINavigationBarDelegate, NavBarLayoutDelegate>
+
+@end
+
+#pragma mark -
+
+@interface OWSNavigationController () <UIGestureRecognizerDelegate>
+
+@end
+
+#pragma mark -
+
+@implementation OWSNavigationController
+
+- (instancetype)initWithRootViewController:(UIViewController *)rootViewController
+{
+    self = [self initWithNavigationBarClass:[OWSNavigationBar class] toolbarClass:nil];
```

Comment:
Don't you want to do the obligatory `if self` dance?

Reply:
danced!